syntheseus/interface/models.py

```diff
@@ -31,7 +31,7 @@ class Config:
     score: Optional[float] = None  # Any other score.
     reaction: Optional[str] = None  # Reaction smiles.
     rxnid: Optional[int] = None  # Template id, if applicable.
-    metadata: Optional[Dict[str, Any]] = {}  # Additional metadata.
+    metadata: Dict[str, Any] = {}  # Additional metadata.
```

Comment:
Are you sure that setting the default to `{}` here avoids creating a shared dict for all instances of this class? I recall that python does this often, and this is why for example people avoid using dicts/lists as default kwargs and why `dataclass` uses `default_factory`. I recall `default_factory` being an issue for `pydantic` but can't remember precisely how this was resolved. Just thought I should flag this as a potential issue.

Reply:
Great investigation Kris! Do you think it would be worth putting a small comment to note this behaviour? It could save us some trouble if we ever switch away from pydantic in the future.

Reply:
Make sense, done now.